index.go

```diff
@@ -181,50 +192,24 @@ func (i *index) Delete(bucket *bbolt.Bucket, ref Reference, doc Document) error
 
 // addRefToBucket adds the reference to the correct key in the bucket. It handles multiple reference on the same location
 func addRefToBucket(bucket *bbolt.Bucket, key Key, ref Reference) error {
-	entryBytes := bucket.Get(key)
-	var entry Entry
-
-	if len(entryBytes) == 0 {
-		entry = EntryFrom(ref)
-	} else {
-		if err := entry.Unmarshal(entryBytes); err != nil {
-			return err
-		}
-		entry.Add(ref)
-	}
-
-	iBytes, err := entry.Marshal()
+	// first check if there's a sub-bucket
+	subBucket, err := bucket.CreateBucketIfNotExists(key)
 	if err != nil {
 		return err
 	}
-
-	return bucket.Put(key, iBytes)
+	subBucket.FillPercent = 0.9
+	return subBucket.Put(ref, []byte{})
 }
 
 // removeRefFromBucket removes the reference from the bucket. It handles multiple reference on the same location
 func removeRefFromBucket(bucket *bbolt.Bucket, key Key, ref Reference) error {
-	entryBytes := bucket.Get(key)
-	var entry Entry
-
-	if len(entryBytes) == 0 {
-		return nil
-	}
-
-	if err := entry.Unmarshal(entryBytes); err != nil {
-		return err
-	}
-	entry.Delete(ref)
-
-	if entry.Size() == 0 {
-		return bucket.Delete(key)
-	}
-
-	iBytes, err := entry.Marshal()
+	// first check if there's a sub-bucket
+	subBucket, err := bucket.CreateBucketIfNotExists(key)
```

Comment:
why create the bucket if the item to remove doesnt exist?

Reply:
changed it to a bucket check, if no bucket return nil.